data/contract_docs/logs.md

```diff
@@ -0,0 +1,97 @@
+# Logs
+
+Status: EARLY Draft/Proposal
+
+## LogLines
+
+### Properties and field requirements
+
+- **Time field** - _required_
+  - There must be at least one non nullable time field
+  - If there are multiple time fields present, following will decide the priority
+    - First matching time field with name `timestamp`
+    - or first matching time field with name `ts`
+    - or first matching time field
+- **Message field** - _required_
+  - There must be at lease one non nullable string field must present
+  - If more than one string fields found, the following will decide the priority
+    - First matching string field with name `body`
+    - or first matching string field with name `message`
+    - or first matching string field
```

Comment:
the first two fields (timestamp & log-line) can be looked up both "by order" (first-of-type) and "by name".

i wonder if we should simplify here, to either make these only "by order", or only "by name".